pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -402,15 +430,15 @@ func (ghs *githubScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 
 			wg1.Add(2)
 			go ghs.getBranches(ctx, genClient, work[i], now, branchCh, &wg1)
-			go ghs.getPullRequests(ctx, genClient, work[i], now, pullRequestCh, &wg1)
+			go getPullRequests(ghs, ctx, genClient, work[i], now, pullRequestCh, &wg1)
 			if ghs.cfg.MetricsBuilderConfig.Metrics.GitRepositoryContributorCount.Enabled {
 				wg1.Add(1)
 				go ghs.getContributorCount(ctx, genClient, work[i], now, &wg1)
 			}
 		}
 
 		for i := 0; i < opBuf; i++ {
-			go ghs.processPullRequests(ctx, genClient, now, pullRequestCh)
+			go processPullRequests(ghs, ctx, genClient, now, pullRequestCh)
```

Comment:
These calls feel off to me and inconsistent with each other. I don't think is actually buffering the channel.

Reply:
Just need to pass the scraper in and they'd look the same.  Not sure how the channel wouldn't be buffered though since we are setting it to a size > 0